openquake/hazardlib/geo/surface/base.py

```diff
@@ -22,10 +22,28 @@
 """
 import numpy
 import math
+from openquake.hazardlib.geo.mesh import Mesh
 from openquake.hazardlib.geo import geodetic, utils, Point, Line,\
     RectangularMesh
 
 
+def _get_finite_mesh(mesh):
+    idx = numpy.isfinite(mesh.lons.flat)
```

Comment:
`idx` is an array of booleans, not of indices. In such I use the convention of calling `ok`:
```python
ok = numpy.isfinite(mesh.lons.flat)
if numpy.all(ok):
￼        return mesh
```